### PR TITLE
MF:384 Vitals Widget Header: Implement different visual states

### PR DESCRIPTION
**Pull request by donaldkibet**
### What does this PR do ?

- Fix exhaustive-deps for biometrics and vitals widgets
- start of vital headers
- Vitals Widget Header: Implement different visual states

### CODE REVIEW

File: src/widgets/vitals/vitals-card.resource.tsx [hunk 1]
```diff
@@ -4,21 +4,24 @@ import {
...
-import { map } from "rxjs/operators";
+import { concatMapTo, map } from "rxjs/operators";
 import { Vitals } from "./vitals-form.component";
 import { FHIRResource } from "../../types/fhir-resource";
 import { ConfigObject } from "../../config-schema";
+import { calculateBMI } from "../heightandweight/heightandweight-helper";
 
 export type PatientVitals = {
-  id: String;
+  id: string;
```
Comment (FlorianRappl) [inline src/widgets/vitals/vitals-card.resource.tsx:14]: Great catch!

Thanks for correcting this :beers:

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks a lot :beers:!

File: src/widgets/vitals/vitals-header/vital-header-details.compont.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
```
Comment (jonathandick) [inline src/widgets/vitals/vitals-header/vital-header-details.compont.tsx]: this file has a spelling error, please rename vital-header-details.component.tsx

**Review by jonathandick — APPROVED** [review]
Looks good other than one small spellig issue

**Comment by FlorianRappl** [comment]
Would love to merge this in, too. Can you take care of the conflicts @donaldkibet ? Thanks a lot - much appreciated 👍

File: src/widgets/vitals/vitals-header/vital-header-details.component.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import styles from "./vital-header-details.component.scss";
+
+interface VitalHeaderStateDetailsProps {
+  unitName: string;
+  value: string | number;
+  unitSymbol: string | React.ReactNode;
```
Comment (FlorianRappl) [inline src/widgets/vitals/vitals-header/vital-header-details.component.tsx]: I think React.ReactChild is essentially `string | ReactNode` (it also contains boolean and number, but that should  not matter much).

File: src/widgets/vitals/vitals-header/vital-header-details.component.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+const VitalHeaderStateDetails: React.FC<VitalHeaderStateDetailsProps> = ({
+  unitName,
+  value,
+  unitSymbol
+}) => {
+  return (
+    <div className={styles.vitalsHeaderStateDetailsContainer}>
+      <label className={styles.label01}>{unitName}</label>
+      <label>
+        <span className={styles.bodyShort02}>{value ? value : "-"} </span>
```
Comment (FlorianRappl) [inline src/widgets/vitals/vitals-header/vital-header-details.component.tsx]: Could also do `value || '-'`.

File: src/widgets/vitals/vitals-header/vital-header-state.component.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+  useEffect(() => {
+    if (patientUuid && config) {
+      const sub = performPatientsVitalsSearch(
+        config.concepts,
+        patientUuid
+      ).subscribe(
+        vitals => {
+          if (vitals.length) {
+            setVitals(vitals[0]);
```
Comment (FlorianRappl) [inline src/widgets/vitals/vitals-header/vital-header-state.component.tsx]: Coupled state. Should we have a more complex state object or even a reducer here? Once we need to use two (or more) `setX` instructions at the same time we know that something is not fully right.

File: src/widgets/vitals/vitals-header/vital-header-title.component.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            {view === "Warning" && (
+              <WarningFilled20
+                title={"WarningFilled"}
+                aria-label="Add"
+                className={styles.warningColor}
+              />
+            )}
+            <span className={styles.vitalName}>Vitals & Biometrics</span>
+            <span className={styles.bodyShort01}>
+              Last recorded:{" "}
```
Comment (FlorianRappl) [inline src/widgets/vitals/vitals-header/vital-header-title.component.tsx]: We would need some translations.

File: src/widgets/vitals/vitals-header/vital-header-title.component.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+import isToday from "dayjs/plugin/isToday";
+import { Button } from "carbon-components-react";
+import { PatientVitals } from "../vitals-card.resource";
+import { isEmpty } from "lodash-es";
+dayjs.extend(isToday);
+
+interface VitalsHeaderStateTitleProps {
+  view: string;
+  vitals: PatientVitals;
+  toggleView: Function;
```
Comment (FlorianRappl) [inline src/widgets/vitals/vitals-header/vital-header-title.component.tsx]: Please don't use `Function`. Type the function instead (e.g., `toggleView(): void`).

**Review by FlorianRappl — COMMENTED** [review]
LGTM - thanks a lot for the great work @donaldkibet !

**Pull request merged**